### DISCUSSION TITLE
Example of using blurhash with next/image

### DISCUSSION THREAD

**Discussion by IHIutch** [vercel/next.js#26161] [Help] [answered]
The [NextJS 11 blog post](https://nextjs.org/blog/next-11#image-placeholders) mentions using [blurha.sh](https://blurha.sh/) as a source for the new `blurDataURL` and `placeholder` props, but blurhash doesn't return a base64 string, which is required for `blurDataURL`. If there is an example of how this might work, I'd be very grateful. 

It's awesome that the new Image component can blur up, but as of this moment, I have no idea how to generate the "blur" this component requires!

**Comment by kristianstroka** [comment]
did somebody tried that? I'm also surprised how to use that

**Comment by styfle** [comment, marked as answer]
I think the author of the blog post accidentally linked to the wrong tool. 

You can't use `bluhash` because it requires canvas, but you can use a tool like `sharp` or `plaiceholder` to generate a base64 image at build time. https://plaiceholder.co/docs/usage#base64

We'll make sure the update the blog post and docs, thanks!

**Reply by stnwk** [reply]
Even with https://plaiceholder.co/docs/usage#base64 the documentation says:

> For a "blurred" effect, add filter: blur(<value>) and transform: scale(<value>) styles to the image.

Really not sure how this works together with the `blurDataURL` api? Probably should be called `placeholderDataURL` 😄
I love the idea but the api provided by nextjs is a bit confusing and the docs don't help much either at the moment.

Would be awesome to see an example for how to integrate it with **plaiceholder** and if possible **blurhash**. Thanks!

**Reply by styfle** [reply]
@stnwk There's an example on that linked page but its hidden behind the "Next.js" tab.

```tsx
import * as React from "react";
import type { InferGetStaticPropsType } from "next";
import Image from "next/image";
import { getPlaiceholder } from "plaiceholder";

export const getStaticProps = async () => {
  const { base64, img } = await getPlaiceholder("/path-to-your-image.jpg");

  return {
    props: {
      imageProps: {
        ...img,
        blurDataURL: base64,
      },
    },
  };
};

const Page: React.FC<InferGetStaticPropsType<typeof getStaticProps>> = ({
  imageProps,
}) => (
  <div>
    <Image {...imageProps} placeholder="blur" />
  </div>
);

export default Page;
```

**Reply by stnwk** [reply]
True, works without having to do `For a "blurred" effect, add filter: blur() and transform: scale() styles to the image.`

Thanks!

**Reply by ivansevillaa** [reply]
Hi @stnwk! if you want to use blurhash, you could use a hook that I've created. It's [here](https://github.com/ivansevillaa/use-next-blurhash.). You just need to pass the hash generated to the hook, and it will return the blurred image to pass to the Image Component.

Any feedback is welcome 😁

**Reply by Fonger** [reply]
> Hi @stnwk! if you want to use blurhash, you could use a hook that I've created. It's [here](https://github.com/ivansevillaa/use-next-blurhash.). You just need to pass the hash generated to the hook, and it will return the blurred image to pass to the Image Component.
> 
> Any feedback is welcome 😁

@ivansevillaa use-next-blurhash doesn't work under server-side rendering. next/image requires blurDataURL at server side

**Reply by ivansevillaa** [reply]
@Fonger using next/dynamic either? https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr

**Reply by BLITZet-Studios** [reply]
> @stnwk There's an example on that linked page but its hidden behind the "Next.js" tab.
> 
> ```tsx
> import * as React from "react";
> import type { InferGetStaticPropsType } from "next";
> import Image from "next/image";
> import { getPlaiceholder } from "plaiceholder";
> 
> export const getStaticProps = async () => {
>   const { base64, img } = await getPlaiceholder("/path-to-your-image.jpg");
> 
>   return {
>     props: {
>       imageProps: {
>         ...img,
>         blurDataURL: base64,
>       },
>     },
>   };
> };
> 
> const Page: React.FC<InferGetStaticPropsType<typeof getStaticProps>> = ({
>   imageProps,
> }) => (
>   <div>
>     <Image {...imageProps} placeholder="blur" />
>   </div>
> );
> 
> export default Page;
> ```


Im Still Learning How to do all of this but Im Slowly Figuring it out, Im Using Scratch.mit.edu to start with basics but Im Working on getting better

**Comment by laclance** [comment]
doesn't work for me 
```
Error: Image with src "/img/bg-sign-in.jpg" has "placeholder='blur'" property but is missing the "blurDataURL" property.
          Possible solutions:
            - Add a "blurDataURL" property, the contents should be a small Data URL to represent the image
            - Change the "src" property to a static import with one of the supported file types: jpeg,png,webp,avif
            - Remove the "placeholder" property, effectively no blur effect
          Read more: https://nextjs.org/docs/messages/placeholder-blur-data-url
